hazelcast/src/main/java/com/hazelcast/internal/bootstrap/TpcServerBootstrap.java

```diff
@@ -0,0 +1,293 @@
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.spi.impl.NodeEngineImpl;
+import com.hazelcast.spi.impl.operationexecutor.impl.AltoOperationScheduler;
+import com.hazelcast.spi.impl.operationexecutor.impl.AltoPartitionOperationThread;
+import com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl;
+import com.hazelcast.spi.properties.HazelcastProperty;
+
+import java.io.UncheckedIOException;
+import java.net.BindException;
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ThreadFactory;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static com.hazelcast.internal.server.ServerContext.KILO_BYTE;
+import static com.hazelcast.internal.tpc.AsyncSocketOptions.SO_KEEPALIVE;
+import static com.hazelcast.internal.tpc.AsyncSocketOptions.SO_RCVBUF;
+import static com.hazelcast.internal.tpc.AsyncSocketOptions.SO_REUSEPORT;
+import static com.hazelcast.internal.tpc.AsyncSocketOptions.SO_SNDBUF;
+import static com.hazelcast.internal.tpc.AsyncSocketOptions.TCP_NODELAY;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+@SuppressWarnings("checkstyle:MagicNumber, checkstyle:")
+public class TpcServerBootstrap {
+    public static final HazelcastProperty ALTO_ENABLED = new HazelcastProperty(
+            "hazelcast.internal.alto.enabled");
+    public static final HazelcastProperty ALTO_EVENTLOOP_COUNT = new HazelcastProperty(
+            "hazelcast.internal.alto.eventloop.count");
+    private static final int TERMINATE_TIMEOUT_SECONDS = 5;
+    private final NodeEngineImpl nodeEngine;
+    private final ILogger logger;
+    private final Address thisAddress;
+    private TpcEngine tpcEngine;
+    @SuppressWarnings("java:S1170")
+    private final boolean tcpNoDelay = true;
+    private final boolean enabled;
+    private final Map<Reactor, Supplier<? extends ReadHandler>> readHandlerSuppliers = new HashMap<>();
+    private final List<AsyncServerSocket> serverSockets = new ArrayList<>();
+    private final Config config;
+    private volatile List<Integer> clientPorts;
+
+    public TpcServerBootstrap(NodeEngineImpl nodeEngine) {
+        this.nodeEngine = nodeEngine;
+        this.logger = nodeEngine.getLogger(TpcServerBootstrap.class);
+        this.config = nodeEngine.getConfig();
+        this.enabled = loadAltoEnabled();
+        this.thisAddress = nodeEngine.getThisAddress();
+    }
+
+    private boolean loadAltoEnabled() {
+        boolean enabled0;
+        String enabledString = nodeEngine.getProperties().getString(ALTO_ENABLED);
+        if (enabledString != null) {
+            enabled0 = Boolean.parseBoolean(enabledString);
+        } else {
+            enabled0 = config.getAltoConfig().isEnabled();
+        }
+        logger.info("TPC: " + (enabled0 ? "enabled" : "disabled"));
```

Comment:
"TPC: " or "Alto: "?

Reply:
AFAIK, tpc is the name of the low level engine (eventloops etc. ) and Alto is the general runtime (data structures etc.). But currently enabling alto only enables tpc as there is no alto runtime yet. IMHO, I would also prefer using one of them but this isn't my decision.